aten/src/ATen/templates/UnambiguousTorchOperators.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+// ${generated_comment}
+
+#include <ATen/Functions.h>
+#include <ATen/Tensor.h>
+
+// Extension writers: do you write wrapper functions? Are you frustrated with
+// resolving overloads of operators? Are you frustrated with dealing with
+// pointer-to-methods and resolving overloads of pointer-to-methods?? Look no
+// further, this is the utility for you.
+//
+// Given an operator schema: aten::op.overload(...
+//
+// Use ATEN_FN2(op, overload) to get a *function* version of the operator
+// that is guaranteed to not be overloaded. This means that you can safely
+// decltype(&ATEN_FN2(op, overload)) it. NB: the 2 means this macro takes 2 args.
+//
+// Given an operator schema without an overload name: aten::op(...
+//
+// Use ATEN_FN(op) to get an unambiguous *function* version of the operator.
+//
+// There is some interesting behavior for out= operations.
+// ATEN_FN2(sin, "out") gives a function that is *faithful* to the schema;
+// that is, the order of arguments is exactly what it looks like in the schema.
+
+#define ATEN_FN2(op_name, overload) at::_ops::op_name##_##overload
+#define ATEN_FN(op_name) at::_ops::op_name
+
+// WARNING: Please do not call any of the ops in the _ops namespace directly.
+// Use the ATEN_FN macros. We do not guarantee stability of the naming
+// scheme for the functions in at::_ops.
+namespace at { namespace _ops {
+
+// NB: this is special cased here because requires_grad_ returns a
+// const Tensor&. The codegen assumes that in-place functions return Tensor&.
+TORCH_API Tensor & requires_grad_(Tensor & self, bool requires_grad);
```

Comment:
What happens when you let the codegen handle `requires_grad_`? The comment says that it returns a const Tensor& but the declaration here just returns a normal Tensor&.

Reply:
If I let the codegen handle `requires_grad_`, it outputs the following definition:
```
Tensor & requires_grad_(Tensor & self, bool requires_grad) {
    return self.requires_grad_(requires_grad);
}
```
Now, this is problematic because `self.requires_grad_(requires_grad)` returns a `const Tensor&`, but `ATEN_FN(requires_grad_)` returns a Tensor&, so the compiler complains that we're coercing a `const Tensor&` to a `Tensor&`. All other in-place functions return non-const references (e.g. `self.sin_()` returns a Tensor&).

`ATEN_FN(requires_grad_)` should probably have a signature that is equal to whatever signature an override for `requires_grad_` should be. build/aten/src/ATen/core/TensorMethods.cpp shows that that signature should be `at::Tensor & (at::Tensor &, bool)`:
```
// aten::requires_grad_(Tensor(a!) self, bool requires_grad=True) -> Tensor(a!)
at::Tensor & Tensor::__dispatch_requires_grad_(bool requires_grad) const {
    static auto op = c10::Dispatcher::singleton()
        .findSchemaOrThrow("aten::requires_grad_", "")
        .typed<at::Tensor & (at::Tensor &, bool)>();
    return op.call(const_cast<Tensor&>(*this), requires_grad);
}
```

Reply:
hmmm ok, so it sounds like all of the auto-generated `inplace` method signatures in `TensorMethods.h` are codegen'd to return `Tensor&`, but you're forced to special case `requires_grad_` because it has a `manual_cpp_binding` with a [different signature](https://github.com/pytorch/pytorch/blob/5ea87f9c24cdc7bfde9e4cacca1985592d65e702/aten/src/ATen/templates/TensorBody.h#L868) that returns `const Tensor&`.

I'm pretty sure that long term, we want to eventually kill `Tensor&` from any C++ signatures and use `const Tensor&` everywhere, but it'd take some work to get the codegen to uniformly do this.

Maybe just include somewhere in the comment that we eventually want to remove that special case and let the codegen handle it, once we remove mutable `Tensor&` returns from the codegen?